File: zoomx/src/main/java/com/zoomx/zoomx/model/InfoModel.java
package com.zoomx.zoomx.model;

/**
 * Created by Ahmed Fathallah on 2/15/2018.
 */

public class InfoModel {
    public static final int TEXT_VIEW_TYPE = 601;
    public static final int SWITCH_VIEW_TYPE = 602;
    public static final int BUTTON_VIEW_TYPE = 603;

    private String title;
    private int type;
    private String value;
    private boolean state;
    private Runnable buttonAction;

    public InfoModel(int type, String title, String value) {
        this.title = title;
        this.type = type;
        this.value = value;
        this.buttonAction = null;
    }

    public InfoModel(int type, String title, String value, Runnable buttonAction) {
        this.title = title;
        this.type = type;
        this.value = value;
        this.buttonAction = buttonAction;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public boolean isState() {
        return state;
    }

    public void setState(boolean state) {
        this.state = state;
    }

    public boolean isTextViewType() {
        return TEXT_VIEW_TYPE == type;
    }

    public boolean isSwitchType() {
        return SWITCH_VIEW_TYPE == type;
    }

    public boolean isButtonType() {
        return BUTTON_VIEW_TYPE == type;
    }

    public Runnable getButtonAction() {
        return buttonAction;
    }
}

File: zoomx/src/main/java/com/zoomx/zoomx/model/InfoModel.kt
package com.zoomx.zoomx.model

/**
 * Created by Ahmed Fathallah on 2/15/2018.
 */

class InfoModel {

    var title: String? = null
    var type: Int = 0
    var value: String? = null
    var isState: Boolean = false
    var buttonAction: Runnable? = null
        private set

    val isTextViewType: Boolean
        get() = TEXT_VIEW_TYPE == type

    val isSwitchType: Boolean
        get() = SWITCH_VIEW_TYPE == type

    val isButtonType: Boolean
        get() = BUTTON_VIEW_TYPE == type

    constructor(type: Int, title: String, value: String) {
        this.title = title
        this.type = type
        this.value = value
        this.buttonAction = null
    }

    constructor(type: Int, title: String, value: String, buttonAction: Runnable) {
        this.title = title
        this.type = type
        this.value = value
        this.buttonAction = buttonAction
    }

    companion object {
        val TEXT_VIEW_TYPE = 601
        val SWITCH_VIEW_TYPE = 602
        val BUTTON_VIEW_TYPE = 603
    }
}
